core/src/main/java/org/apache/spark/util/collection/unsafe/sort/PrefixComparators.java

```diff
@@ -69,49 +100,161 @@ public static long computePrefix(double value) {
+    }
+  }
+
+  // unsigned desc null first
+  public static final class UnsignedPrefixComparatorDescNullFirst extends RadixSortSupport {
+    @Override public boolean sortDescending() { return true; }
+    @Override public boolean sortSigned() { return false; }
+    @Override public NullOrder nullOrder() { return NullOrder.FIRST; }
+    public int compare(long bPrefix, long aPrefix) {
+      if (aPrefix == 0L && bPrefix == 0L) {
+        return 0;
+      }
+      if (bPrefix == 0L) {
+        return -1;
+      } else if (aPrefix == 0L) {
+        return 1;
+      }
+      return UnsignedLongs.compare(aPrefix, bPrefix);
+    }
+  }
+
+  // unsigned desc null last (default)
   public static final class UnsignedPrefixComparatorDesc extends RadixSortSupport {
     @Override public boolean sortDescending() { return true; }
     @Override public boolean sortSigned() { return false; }
-    @Override
+    @Override public NullOrder nullOrder() { return NullOrder.LAST; }
     public int compare(long bPrefix, long aPrefix) {
+      if (aPrefix == 0L && bPrefix == 0L) {
+        return 0;
+      }
+      if (bPrefix == 0L) {
+        return 1;
+      } else if (aPrefix == 0L) {
+        return -1;
+      }
       return UnsignedLongs.compare(aPrefix, bPrefix);
     }
   }
 
+  // signed asc null first (default)
   public static final class SignedPrefixComparator extends RadixSortSupport {
     @Override public boolean sortDescending() { return false; }
     @Override public boolean sortSigned() { return true; }
-    @Override
+    @Override public NullOrder nullOrder() { return NullOrder.FIRST; }
+    public int compare(long a, long b) {
+        if (a == Long.MIN_VALUE && b == Long.MIN_VALUE) {
+          return 0;
+        }
+        if (a == Long.MIN_VALUE) {
+          return -1;
+        } else if (b == Long.MIN_VALUE) {
+          return 1;
+        }
+      return (a < b) ? -1 : (a > b) ? 1 : 0;
+    }
+  }
+
+  // signed asc null last
+  public static final class SignedPrefixComparatorNullLast extends RadixSortSupport {
+    @Override public boolean sortDescending() { return false; }
+    @Override public boolean sortSigned() { return true; }
+    @Override public NullOrder nullOrder() { return NullOrder.LAST; }
     public int compare(long a, long b) {
+      if (a == Long.MIN_VALUE && b == Long.MIN_VALUE) {
+        return 0;
+      }
+      if (a == Long.MIN_VALUE) {
+        return 1;
+      } else if (b == Long.MIN_VALUE) {
+        return -1;
+      }
+      return (a < b) ? -1 : (a > b) ? 1 : 0;
+    }
+  }
+
+  // signed desc null first
+  public static final class SignedPrefixComparatorDescNullFirst extends RadixSortSupport {
+    @Override public boolean sortDescending() { return true; }
+    @Override public boolean sortSigned() { return true; }
+    @Override public NullOrder nullOrder() { return NullOrder.FIRST; }
+    public int compare(long b, long a) {
+        if (a == Long.MIN_VALUE && b == Long.MIN_VALUE) {
```

Comment:
The fact that MIN_VALUE could be also literal min value was also my concern too. I will check what I can in SortOrder.scala. Thanks!